src/app/api/v1/endpoints/extract_data.py

```diff
@@ -0,0 +1,290 @@
+            coordinates=Coordinates(
+                east=coord.east.coordinate_value,
+                north=coord.north.coordinate_value,
+                projection=srs,
+            ),
+        )
+
+    coord_extractor = CoordinateExtractor(pdf_page)
+    extracted_coord = coord_extractor.extract_coordinates_from_bbox(
+        pdf_page, extract_data_request.page_number, user_defined_bbox
+    )
+
+    if isinstance(extracted_coord, LV03Coordinate):
+        return create_response(extracted_coord, "LV03")
+
+    if isinstance(extracted_coord, LV95Coordinate):
+        return create_response(extracted_coord, "LV95")
+
+    return None
+
+
+def extract_elevation(
+    extract_data_request: ExtractDataRequest, pdf_page: fitz.Page, user_defined_bbox: fitz.Rect
+) -> ExtractDataResponse:
+    """Extract the elevation from a PNG image.
+
+    Args:
+        extract_data_request (ExtractDataRequest): The request data.
+        pdf_page (fitz.Page): The PDF page.
+        user_defined_bbox (fitz.Rect): The user-defined bounding box.
+
+    Returns:
+        ExtractDataResponse: The extracted elevation.
+    """
+    elevation_extractor = ElevationExtractor(pdf_page)
+    elevation = elevation_extractor.extract_elevation_from_bbox(
+        pdf_page, extract_data_request.page_number, user_defined_bbox
+    )
+
+    if elevation is None:
+        return NotFoundResponse(
+            detail="Elevation not found.",
+            bbox=extract_data_request.bbox,
+        )
+
+    bbox = BoundingBox(
+        x0=elevation.rect.x0,
+        y0=elevation.rect.y0,
+        x1=elevation.rect.x1,
+        y1=elevation.rect.y1,
+    )
+    return ExtractElevationResponse(
+        bbox=bbox,
+        elevation=elevation.elevation,
+    )
+
+
+def extract_text(pdf_page: fitz.Page, user_defined_bbox: fitz.Rect) -> ExtractDataResponse:
+    """Extract text from a PNG image.
+
+    Args:
+        extract_data_request (ExtractDataRequest): The request data.
+        pdf_page (fitz.Page): The PDF page.
+        user_defined_bbox (fitz.Rect): The user-defined bounding box.
+
+    Returns:
+        ExtractDataResponse: The extracted text.
+    """
+    # Extract the text
+    text_lines = extract_text_lines_from_bbox(pdf_page, user_defined_bbox)
+
+    # Convert the text lines to a string
+    text = ""
+    text_based_bbox = fitz.Rect()
+    for text_line in text_lines:
+        text += text_line.text + " "
+        text_based_bbox = text_based_bbox + text_line.rect
+
+    bbox = BoundingBox.load_from_fitz_rect(text_based_bbox)
+    return ExtractTextResponse(bbox=bbox, text=text)
+
+
+def extract_number(pdf_page: fitz.Page, user_defined_bbox: fitz.Rect):
+    """Extract numbers from a PNG image.
+
+    Args:
+        extract_data_request (ExtractDataRequest): The request data.
+        pdf_page (fitz.Page): The PDF page.
+        user_defined_bbox (fitz.Rect): The user-defined bounding box.
+
+    Returns:
+        ExtractDataResponse: The extracted numbers.
```

Comment:
we extract only one number per request, correct?

Reply:
Good catch